rust/arrow/src/array/array_binary.rs

```diff
@@ -162,6 +163,16 @@ impl<OffsetSize: BinaryOffsetSizeTrait> GenericBinaryArray<OffsetSize> {
     }
 }
 
+impl<'a, OffsetSize: BinaryOffsetSizeTrait> TypedArray
+    for &'a GenericBinaryArray<OffsetSize>
```

Comment:
I had initially hoped to make TypedArray an extension of the Array trait.  But, to use a lifetime in the 'type' declarations, I found that I needed to either make TypedArray include a generic lifetime (TypedArray<'a>), or I needed to implement it for &'a TypedArray.

My previous attempts at this tried to use `trait TypedArray<'a> : Array`, but once I tried using it on the comparison kernels ' I hit on lifetime issues trying to prove the equivalence of NativeTypes... so I'm keeping it simple for the experiment.